File: frontend/src/types/document.ts
import type { Span } from "@/types/span";

export type AppMode = "trust" | "batch" | "correction";

export interface AppDocument {
  id: string;
  filename: string;
  text: string;
  uploaded_at: string;
}

export interface AnalysisRequest {
  text: string;
  filename: string;
  mode: AppMode;
}

export interface AnalysisResult {
  session_id: string;
  document: AppDocument;
  spans: Span[];
  mode: AppMode;
}

File: backend/models/document.py
from typing import Literal

from pydantic import BaseModel

from models.span import Span


class Document(BaseModel):
    id: str
    filename: str
    text: str
    uploaded_at: str


class AnalysisRequest(BaseModel):
    text: str
    filename: str
    mode: Literal["trust", "batch", "correction"]


class AnalysisResult(BaseModel):
    session_id: str
    document: Document
    spans: list[Span]
    mode: Literal["trust", "batch", "correction"]
